js/gateio.js

```diff
@@ -2256,6 +2256,125 @@ module.exports = class gateio extends Exchange {
         };
     }
 
+    parsePosition (position, market = undefined) {
+        //
+        //     {
+        //         value: "12.475572",
+        //         leverage: "0",
+        //         mode: "single",
+        //         realised_point: "0",
+        //         contract: "BTC_USDT",
+        //         entry_price: "62422.6",
+        //         mark_price: "62377.86",
+        //         history_point: "0",
+        //         realised_pnl: "-0.00624226",
+        //         close_order:  null,
+        //         size: "2",
+        //         cross_leverage_limit: "25",
+        //         pending_orders: "0",
+        //         adl_ranking: "5",
+        //         maintenance_rate: "0.005",
+        //         unrealised_pnl: "-0.008948",
+        //         user: "6645677",
+        //         leverage_max: "100",
+        //         history_pnl: "14.98868396636",
+        //         risk_limit: "1000000",
+        //         margin: "0.740721495056",
+        //         last_close_pnl: "-0.041996015",
+        //         liq_price: "59058.58"
+        //     }
+        //
+        const contract = this.safeValue (position, 'contract');
+        market = this.safeMarket (contract, market);
+        const now = this.milliseconds ();
+        const maintenance_rate = this.safeValue (position, 'maintenance_rate');
+        // const markPrice = this.safeValue (position, 'mark_price');
+        // const contractSize = this.safeValue (position, 'size');
+        const notional = this.safeValue (position, 'value');
+        return {
+            'info': position,
+            'symbol': market['symbol'],
+            'timestamp': now,
+            'datetime': this.iso8601 (now),
+            'initialMargin': this.safeValue (position, 'margin'),
+            'initialMarginPercentage': undefined,
+            'maintenanceMargin': maintenance_rate * notional,
+            'maintenanceMarginPercentage': maintenance_rate * 100,
+            'entryPrice': this.safeValue (position, 'entry_price'),
+            'notional': notional,
+            'leverage': this.safeValue (position, 'leverage'),
+            'unrealizedPnl': this.safeValue (position, 'unrealised_pnl'),
+            'contracts': undefined,
+            'contractSize': this.safeValue (position, 'size'),
+            //     realisedPnl: position['realised_pnl'],
+            'marginRatio': undefined,
+            'liquidationPrice': this.safeValue (position, 'liq_price'),
+            'markPrice': this.safeValue (position, 'mark_price'),
+            'collateral': this.safeValue (position, 'margin'),
+            'marginType': undefined,
+            'side': undefined,
+            'percentage': undefined,
```

Comment:
What is this a percentage of?

Reply:
it's the percentage of unrealizedPnl / initialMargin